### ISSUE TITLE
Clusterloader2 use existing prometheus?

### ISSUE THREAD

**Issue by aakarshg** [kubernetes/perf-tests#1057]
Hello Some of the distros like OpenShift come with a prometheus installation on the cluster, is there anyway we can use this prometheus server's metrics instead of trying to stand up another one, and then have node exporter running ?

**Comment by wojtek-t** [comment]
@mm4tt @mborsz @oxddr

**Comment by oxddr** [comment]
At the moment I don't this is possible. But the reason for that is we have some assumptions backed in the clusterloader code (@mm4tt correct me if I am wrong). This should be matter of a single PR to change how the current setup Prometheus works.

**Comment by mm4tt** [comment]
Depending on what you mean by "prometheus installation".

Our setup is based on [prometheus-operator](https://github.com/coreos/prometheus-operator), it Prometheus-operator is avaiable in the cluster then it should be doable to have cl2 to use it instead of spinning up a new one. But as @oxddr wrote it would require some code changes and it's not possible as of now.

But taking a step back, what is the problem you're trying to solve? It shouldn't be an issue to have 2 (or more) Prometheus servers running in cluster simultaneously.

**Comment by aakarshg** [comment]
If a small change will let us use existing prometheus-operator on cluster, we would at least like to see what the change looks like so that we can apply the change locally to test if it'll work..

We're currently evaluating clusterloader2 for our needs. We're hitting all sorts of problems with the clusterloader2 trying to deploy its own prometheus-operator pods, some of them are privilege related, some of them oddly are due to prometheus-operator launched through clusterloader2 trying to use alertmanager pods in another namespace ( because we've an existing prom stack deployment ). I can try to look into creating another set of manifests specifically for Openshift but all of this seems like doubling the effort given that we already have a prometheus stack on the cluster.

**Comment by mm4tt** [comment]
I see, deploying two prometheus-operators is definitely not a good idea. 

What we could try to do is to extract the logic we have for setting up prometheus operator and add an ability in cl2 to call it (or not call it) conditionally. If prometheus operator is already present in the cluster then it should be doable for cl2 to re-use it. I'm not sure if it will work out of the box as we might have some baked in assumptions about prometheus in our code. But it's definitively worth trying.

Unfortunately, we don't have any free cycles currently to do it. I'd appreciate if you could take a shot at it.
Basically it translates to splitting [coreManifests](https://github.com/kubernetes/perf-tests/blob/3b2fdc539fb14ae7bb64e71370868fb1eb421e0b/clusterloader2/pkg/prometheus/prometheus.go#L44) into two parts, "prometheus-operator" manifests (I think they all start with `0`) and "prometheus-manifests". Then add a flag that allows to not apply prometheus-operator manifests. If the flag is defaulted to "apply operator manifests" then the change should be no-op for our existing CI/CD tests and hopefully will allow you to push forward your OpenShift tests.

Let me know in case of any questions.

**Comment by JanetSch** [comment]
We are also getting issues with using clusterloader2 with OpenShift due to 2 prometheus-operators being deployed. Being able to use the one already there in OpenShift would be a big help.
